File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/structure/ValidationRunStatus.java
package net.nemerosa.ontrack.model.structure;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor(access = AccessLevel.PROTECTED)
public class ValidationRunStatus {

    private final Signature signature;
    private final ValidationRunStatusID statusID;
    private final String description;

    public static ValidationRunStatus of(Signature signature, ValidationRunStatusID validationRunStatusID, String description) {
        return new ValidationRunStatus(
                signature,
                validationRunStatusID,
                description
        );
    }
}

File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/structure/ValidationRunStatus.kt
package net.nemerosa.ontrack.model.structure

import com.fasterxml.jackson.annotation.JsonProperty

class ValidationRunStatus(
        override val id: ID,
        val signature: Signature,
        val statusID: ValidationRunStatusID,
        val description: String?
) : Entity {

    @JsonProperty("passed")
    val isPassed: Boolean = statusID.isPassed

}
